plugin/src/main/scala/org/virtuslab/akkasaferserializer/SaferSerializerPluginComponent.scala

```diff
@@ -0,0 +1,64 @@
+package org.virtuslab.akkasaferserializer
+
+import scala.tools.nsc.plugins.PluginComponent
+import scala.tools.nsc.{Global, Phase}
+
+class SaferSerializerPluginComponent(val pluginOptions: PluginOptions, val global: Global) extends PluginComponent {
+  import global._
+  override val phaseName: String = "akka-safer-serializer-gather"
+  override val runsAfter: List[String] = List("refchecks")
+
+  var annotatedTraitsCache: List[Type] = List()
```

Comment:
Async paranoia: any risk this collection might be modified concurrently by two threads? esp. when compiler is run with `-Ybackend-parallelism` > 1?

Reply:
According to [this article](https://kubuszok.com/2018/speed-up-things-in-scalac-and-sbt/) `-Ybackend-parallelism` is responsible for byte code generation.
And even if, a race condition in our scenario is of no consequence.

Reply:
Okay, good to know, I thought more stuff is parallelized with this flag